baseapp/baseapp.go

```diff
@@ -737,14 +739,26 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 			// additional fee events
 			if len(feeEvents) > 0 {
 				// append the fee events at the end of the other events, since they get charged at the end of the Tx
-				result.Events = append(result.Events, feeEvents.ToABCIEvents()...)
+				result.Events = AggregateEvents(app, result.Events, feeEvents.ToABCIEvents())
 			}
 		}
 	}
 
 	return gInfo, result, anteEvents, priority, ctx, err
 }
 
+// AggregateEvents aggregation logic of result events (ante and postHander events) with feeEvents
+func AggregateEvents(app *BaseApp, resultEvents []abci.Event, feeEvents []abci.Event) []abci.Event {
```

Comment:
It might be helpful to return the error or return some type of success flag. This would make debugging a bit easier to tell when we hit line 755.